src/main/java/use_case/delete_post/DeletePostInputBoundary.java

```diff
@@ -0,0 +1,16 @@
+package use_case.delete_post;
+
+/**
+ * This interface for the DeletePost use case.
+ */
+public interface DeletePostInputBoundary {
+
+    /**
+     * Executes the delete Post use case.
+     * @param DeletePostInputData the input data for the delete Post operation,
+     *        typically containing the ID of the Post to be deleted.
+     */
+    void deletePost(DeletePostInputData DeletePostInputData);
+
+    void switchToDeletePostView();
```

Comment:
I think to make things simple, let's just return to the home screen after deletion. so, you'll likely need a homePagePresenter or smth. Since we don't have that in the code base, you can delete this for now, and someone else can add it in the future when they need it.